featurebyte/service/target_helper/base_feature_or_target_computer.py

```diff
@@ -191,3 +179,84 @@ async def compute(
             validation_parameters=validation_parameters,
         )
         await self.query_executor.execute(params)
+
+    async def _compute_data_frame(
+        self,
+        observation_set: pd.DataFrame,
+        compute_request: ComputeRequestT,
+        output_table_details: TableDetails,
+    ) -> None:
+        validation_parameters = await self.get_validation_parameters(compute_request)
+        request_column_names = set(observation_set.columns)
+        await self._execute(
+            observation_set=observation_set,
+            compute_request=compute_request,
+            output_table_details=output_table_details,
+            validation_parameters=validation_parameters,
+            request_column_names=request_column_names,
+        )
+
+    async def _compute_observation_table(
+        self,
+        observation_set: ObservationTableModel,
+        compute_request: ComputeRequestT,
+        output_table_details: TableDetails,
+    ) -> None:
+        validation_parameters = await self.get_validation_parameters(compute_request)
+        request_column_names = {col.name for col in observation_set.columns_info}
+
+        feature_cache, nodes_hash_dict = await self.feature_table_cache_service.filter_nodes(
```

Comment:
The feature table cache workflow is applicable only to newer observation tables with row index column. That can be checked by `observation_set.has_row_index`. Otherwise, it needs to fallback to the non-cached workflow.